private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/MavenResolver.java

```diff
@@ -189,10 +188,30 @@ public ResolutionResult resolve(ResolutionRequest request) {
             new CompoundListener(consoleLog, errors, coordsListener),
             request.getLocalCache());
 
-    consoleLog.setPhase("Gathering direct dependency coordinates");
-    List<DependencyNode> directDependencies =
+    List<Dependency> depsWithGlobalExclusions = addGlobalExclusions(globalExclusions, dependencies);
+    consoleLog.setPhase(
+        "Resolving and fetching the transitive closure of "
+            + depsWithGlobalExclusions.size()
+            + " artifact(s)");
+    List<DependencyNode> resolvedDependencies =
         resolveBaseDependencies(
-            system, session, repositories, fakeRoot, managedDependencies, amendedDeps);
+            system, session, repositories, fakeRoot, managedDependencies, depsWithGlobalExclusions);
+
+    if (System.getenv("RJE_VERBOSE") != null) {
+      DependencyGraphDumper graphDumper = new DependencyGraphDumper(System.out::println);
+      System.out.println("\nDependency Graph:");
```

Comment:
Please log to `syserr` instead of `sysout`. The output JSON is read from `sysout`, and doesn't expect anything else to be there.

Reply:
You may also want to use `listener.onEvent(new LogEvent(string))` so that the output isn't lost if we're using the fancy UI.

Reply:
I updated it to use `syserr` since the Ansi UI logging is disabled when RJE_VERBOSE is set.